src/background/stores/tab-store.ts

```diff
@@ -22,6 +23,7 @@ export class TabStore extends PersistentStore<TabStoreData> {
         logger: Logger,
         tabId: number,
         persistStoreData: boolean,
+        private readonly urlParser: UrlParser = new UrlParser(),
```

Comment:
Should this be piped from Background -> TabContextFactory -> TabContextStoreHub -> TabStore instead?

It feels weird to create a UrlParser object for every tab, even though that piping feels cumbersome.